src/VulkanSample.cpp

```diff
@@ -2306,7 +2319,7 @@ static void DrawFrame()
 
     // Acquire swapchain image
     uint32_t imageIndex = 0;
-    VkResult res = vkAcquireNextImageKHR(g_hDevice, g_hSwapchain, UINT64_MAX, g_hImageAvailableSemaphore, VK_NULL_HANDLE, &imageIndex);
+    VkResult res = vkAcquireNextImageKHR(g_hDevice, g_hSwapchain, UINT64_MAX, g_hImageAvailableSemaphores.at(g_SwapchainImageIndex), VK_NULL_HANDLE, &imageIndex);
```

Comment:
Maybe a comment on this here: We can't do

```cpp
uint32_t imageIndex = 0;
VkResult res = vkAcquireNextImageKHR(g_hDevice, g_hSwapchain, UINT64_MAX, g_hImageAvailableSemaphores.at(imageIndex), VK_NULL_HANDLE, &imageIndex);
```

because `imageIndex` is written to by `vkAcquireNextImageKHR` (pass by reference), so we can't use is in the call directly.
I had to track the swapchain image index manually, as can be seen at the end of `DrawFrame`.